libraries/commerce/cart/actions/addProductsToCart.js

```diff
@@ -6,43 +6,58 @@ import addProductsToCart from '../action-creators/addProductsToCart';
 import successAddProductsToCart from '../action-creators/successAddProductsToCart';
 import errorAddProductsToCart from '../action-creators/errorAddProductsToCart';
 import setCartProductPendingCount from '../action-creators/setCartProductPendingCount';
-import { getProductPendingCount } from '../selectors';
+import { getProductPendingCount, getAddToCartMetadata } from '../selectors';
+import { getProductById } from '../../product/selectors/product';
 import { messagesHaveErrors } from '../helpers';
 
 /**
- * Adds a product to the cart.
+ * Adds a product to the cart. Includes metadata if available and not provided before.
  * @param {Array} productData The options for the products to be added.
  * @return {Function} A redux thunk.
  */
 const addToCart = productData => (dispatch, getState) => {
-  const pendingProductCount = getProductPendingCount(getState());
+  const state = getState();
+  const pendingProductCount = getProductPendingCount(state);
+  const products = productData.map(product => {
+    const { productData } = getProductById(state, product.productId);
+    const productId = productData.baseProductId || product.productId;
```

Comment:
Please add a comment that you want to fallback to the parent meta data if nothing else is found (i think this is a super edge case though and just working with product.productId would be enough)

Reply:
done